### PR TITLE
context: validate prefix lookups for WithRequest/WithResponse contexts

### PR DESCRIPTION
**Pull request by stevvooe**
Signed-off-by: Stephen J Day <stephen.day@docker.com>

### CODE REVIEW

**Comment by stevvooe** [comment]
cc @dmp42

File: context/http.go [hunk 1]
```diff
@@ -133,6 +133,10 @@ func (ctx *httpRequestContext) Value(key interface{}) interface{} {
 			return ctx.r
 		}
 
+		if !strings.HasPrefix(keyStr, "http.request.") {
+			goto fallback
```
Comment (endophage) [inline context/http.go]: do we have to use `goto`s?

```go

return fallback(key)

...

func fallback(key) {
  return ctx.Context.Value(key)
}
```

**Comment by dmp42** [comment]
LGTM

File: context/http.go (hunk 1 above)
Comment (dmp42) [inline context/http.go]: Ah! :) "goto considered harmful"? we were discussing that with @stevvooe

**Pull request merged**